src/transformers/models/mixtral/modeling_mixtral.py

```diff
@@ -107,15 +107,16 @@ def load_balancing_loss_func(gate_logits: torch.Tensor, num_experts: torch.Tenso
     selected_experts = selected_experts.reshape(-1)
 
     expert_mask = torch.nn.functional.one_hot(selected_experts, num_experts)
+    expert_mask = expert_mask.reshape(-1,top_k, num_experts)
```

Comment:
this is not needed if we just remove `selected_experts = selected_experts.reshape(-1)`